### ISSUE TITLE
Keyboard modifications not working every time I change audio device

### ISSUE THREAD

**Issue by jason-wihardja** [zenangst/KeyboardCowboy#520]
I have my headphone plugged into my MBP, and I have keyboard cowboy modify my keyboard inputs. Everytime I click change audio device (when sometimes I put my headphone down and listen to audio directly through the speakers), none of the keyboard modifications work anymore. I had to quit the app and relaunch it to get it back working.

![image](https://github.com/zenangst/KeyboardCowboy/assets/3488460/619a6117-79ed-42ad-ba28-856b5a645d80)

**Comment by zenangst** [comment]
Oh, that is an interesting bug.
Mind sharing how you’ve setup your workflows so I can try and reproduce it?

**Comment by jason-wihardja** [comment]
I have a macropad, which fires keyboard keys F13 through F20, then have keyboard cowboys basically capture the key for each app like this one for example
![image](https://github.com/zenangst/KeyboardCowboy/assets/3488460/c7aa4055-fff0-4848-b9c7-7c540e926e7d)

**Comment by zenangst** [comment]
Thanks for the additional info @jason-wihardja 

Is the application entirely unresponsive or is it that it doesn't pick up on any more keystrokes?

(I'm trying to narrow it down it the mach port timed out or if there is a stray task lurking around causing issues)

**Comment by jason-wihardja** [comment]
No, it's entirely functional. When I open the app, I can also click through the UI. It's just that the keystrokes are not being processed

**Comment by zenangst** [comment]
Interesting, that leads me to believe that something is happening inside the mach port part of the application. I'll keep digging. Thanks for following up @jason-wihardja

**Comment by zenangst** [comment]
Hey @jason-wihardja, I found one occasion where it would stop working which I have fixed on `main`. I'm not 100% sure that this is the issue that you are running into, but hopefully it is. I'll let you know when the new version is out.

**Comment by jason-wihardja** [comment]
Thanks for the quick fix @zenangst. I'll surely try it out when it is released.

**Comment by jason-wihardja** [comment]
Hi @zenangst, unfortunately it didn't fix the problem. I was wondering, were you able to reproduce the problem in your environment? Because I also start to wonder, whether it has something to do with the key I'm assigning my buttons to, as F13-F20 is not very commonly used

**Comment by zenangst** [comment]
Unfortunately not, I only use the built-in keyboard or an external Apple wireless keyboard and those buttons don't exist.

I might be able to get some help to reproduce it by asking someone at work that have a more "high-tech" keyboard than I have.

**Comment by zenangst** [comment]
Hey @jason-wihardja, I just pushed a new update that improve the handling of F-keys. Mind taking it for a spin to see if it helps in any way?

https://github.com/zenangst/KeyboardCowboy/releases/tag/3.24.2

**Comment by jason-wihardja** [comment]
It didn't work even after I disabled the built-in macos functions, the keys seem to still getting mapped to decrease/increase brightness

![image](https://github.com/zenangst/KeyboardCowboy/assets/3488460/1e1d41d4-86c4-4491-8c2b-bb86aa8bbbf3)

**Comment by jason-wihardja** [comment]
Upon further inspection, I noticed that using the latest version, only F14 and F15 are broken. The other function buttons now work correctly. I tried to tinker with different combinations of the MacOS settings and Keyboard Cowboy settings, but none of the combination worked correctly. A quick Google search gives me several results from other Github repos with similar issues. It seems that there's some sort of bug/issue within the OS itself that causes these 2 buttons to get overridden with the default settings. I'm currently remapping those 2 buttons to other unused shortcuts as a workaround and everything works fine for the time being
